### PR TITLE
docs(libflux): document `fluxcore::semantic::bootsrap` module

### PR DESCRIPTION
**Pull request by pierwill**
Removes `#![allow(missing_docs)]`.

The meaning of this module was not entirely clear to me. Can we say more about what "bootstrap" means here? 🤔

### CODE REVIEW

**Comment by pierwill** [comment]
The `flux` crate calls `bootstrap::infer_stdlib()` [here](https://github.com/influxdata/flux/blob/dbc8181d5a51bc2b5bcb0e7bb921a575435f7013/libflux/flux/build.rs#L51).

File: libflux/flux-core/src/semantic/bootstrap.rs [hunk 1]
```diff
@@ -223,7 +224,7 @@ fn dependencies<'a>(
     }
 }
 
-// Constructs a polytype, or more specifically a generic row type, from a hash map
+/// Constructs a polytype, or more specifically a generic row type, from a hash map.
```
Comment (pierwill) [inline libflux/flux-core/src/semantic/bootstrap.rs:227]: s/row/record? 🤔

File: libflux/flux-core/src/semantic/bootstrap.rs [hunk 2]
```diff
@@ -223,7 +224,7 @@ fn dependencies<'a>(
...
 pub fn build_polytype(from: PolyTypeMap, f: &mut Fresher) -> Result<PolyType, Error> {
     let (r, cons) = build_row(from, f);
```
Comment (pierwill) [inline libflux/flux-core/src/semantic/bootstrap.rs:229]: Thinking of renaming `build_row` to `build_record` in this PR as well...

**Comment by jsternberg** [comment]
One caveat:

> Example: The flux crate calls bootstrap::infer_stdlib(). Does this make sense with my description of the module as "Flux start-up"?

Bootstrap happens at compile time and doesn't happen at runtime. The bootstrap infers the types and is used to embed the results into the resulting library so they can be accessed more quickly.

**Pull request merged**